src/HDWallet.cpp

```diff
@@ -117,6 +117,8 @@ std::optional<PublicKey> HDWallet::getPublicKeyFromExtended(const std::string &e
         return PublicKey(Data(node.public_key, node.public_key + 33), TWPublicKeyTypeSECP256k1);
     case TWCurveED25519:
         return PublicKey(Data(node.public_key, node.public_key + 33), TWPublicKeyTypeED25519);
+    case TWCurveED25519Blake2bNano:
+        return PublicKey(Data(node.public_key, node.public_key + 32), TWPublicKeyTypeED25519Blake2b);
```

Comment:
Not entirely sure if this is correct and the above for regular ed25519 is slightly off, or if this is wrong and the above is correct. In the above (ed25519-sha512) case the length check gets triggered in the constructor as the data length is 33 bytes, but node.public_key[1..33] slice is used for the public key.

Reply:
trezor-crypto uses 0x01 as prefix for ed25519. That is why I originally followed suit. This looks good to me.

Reply:
Actually yours seems to be wrong, see my other comment.

Reply:
> trezor-crypto uses 0x01 as prefix for ed25519. That is why I originally followed suit. This looks good to me.

Ahh, ok. Didn't know that.

Fixed up the commit to pass in all of the 33 bytes here and to handle the 33 bytes correctly in the constructor.